test/v2/connectors/AmendForeignPropertyPeriodSummaryConnectorSpec.scala

```diff
@@ -78,33 +77,57 @@ class AmendForeignPropertyPeriodSummaryConnectorSpec extends ConnectorSpec {
     foreignNonFhlProperty = Some(Seq(foreignNonFhlPropertyEntry))
   )
 
-  private val request: AmendForeignPropertyPeriodSummaryRequest = AmendForeignPropertyPeriodSummaryRequest(
-    nino = Nino(nino),
-    businessId = businessId,
-    taxYear = TaxYear.fromMtd(taxYear),
-    submissionId = submissionId,
-    body = requestBody
-  )
-
   trait Test {
     _: ConnectorTest =>
-    val connector: AmendForeignPropertyPeriodSummaryConnector = new AmendForeignPropertyPeriodSummaryConnector(
+    def taxYear: TaxYear
+
+    protected val connector: AmendForeignPropertyPeriodSummaryConnector = new AmendForeignPropertyPeriodSummaryConnector(
       http = mockHttpClient,
       appConfig = mockAppConfig
     )
+
+    protected val request: AmendForeignPropertyPeriodSummaryRequest = AmendForeignPropertyPeriodSummaryRequest(
+      nino = Nino(nino),
+      businessId = businessId,
+      taxYear = taxYear,
+      submissionId = submissionId,
+      body = requestBody
+    )
   }
 
   "AmendForeignPropertyPeriodSummaryConnector" must {
     "send a request and return 204 no content" in new IfsTest with Test {
+      def taxYear: TaxYear = TaxYear.fromMtd("2019-20")
+
       val outcome = Right(ResponseWrapper(correlationId, ()))
 
       willPut(
           url = s"$baseUrl/income-tax/business/property/periodic?" +
-            s"taxableEntityId=$nino&taxYear=2022-23&incomeSourceId=$businessId&submissionId=$submissionId",
+            s"taxableEntityId=$nino&taxYear=${taxYear.asMtd}&incomeSourceId=$businessId&submissionId=$submissionId",
```

Comment:
Can we factor all this out so we have `stubHttpResponse` and `stubTysHttpResponse` like has been done [here](https://github.com/hmrc/property-business-api/blob/d5aea1e947e82200b8b797017acdffdc47ac0ab9/test/v2/connectors/AmendUkPropertyPeriodSummaryConnectorSpec.scala)